src/QuadVelocityController.cpp

```diff
@@ -31,27 +33,33 @@ void QuadVelocityController::setTargetVelocity(geometry_msgs::Twist twist)
 {
     throttle_pid_.setSetpoint(twist.linear.z);
     pitch_pid_.setSetpoint(twist.linear.x);
-    roll_pid_.setSetpoint(twist.linear.y);
+    roll_pid_.setSetpoint(-twist.linear.y);
     yaw_pid_.setSetpoint(twist.angular.z);
 }
 
 // Needs to be called at regular intervals in order to keep catching the latest velocities.
-iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update()
+iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update(const ros::Time& time)
 {
-    // Get the time delta
     geometry_msgs::Vector3 velocities;
     getVelocities(velocities);
-    double time_delta = 0.0;
+
+    // Get the time delta
+    if (last_time_ == ros::Time(0)) {
+        last_time_ = time;
```

Comment:
I think this should be restructured to:

`if(last_time !-= ros::Time(0))
{ // do normal pid
}
else
{
// return zerod out uav_command
}`

What we do know will automatically have a throttle value which I feel is slightly unsafe because we don't have a valid delta yet.